samples/matter/weather_station/README.rst

```diff
@@ -0,0 +1,154 @@
+   :rows: thingy53_nrf5340
+
+To program and gather logs from the Thingy:53 device, you need the external J-Link programmer.
+If you own an nRF5340 DK that has an on-board J-Link programmer, you can also use it for this purpose.
+
+For remote testing scenarios, if you want to commission the lock device and :ref:`control it remotely <matter_lock_sample_network_mode>` through a Thread network, you also need the following:
+
+* A Matter controller device :ref:`configured on PC or smartphone <ug_matter_configuring>`.
+
+.. note::
+    |matter_gn_required_note|
+
+Overview
+********
+
+The sample uses buttons for changing the lock and device states, and LEDs to show the state of these changes.
+It can be tested in the following ways:
+
+* Standalone, by using a single DK that runs the door lock application.
+* Remotely over the Thread protocol, which requires more devices.
+
+The remote control testing requires a Matter controller, which can be configured either on PC or mobile (for remote testing in a network).
+Both methods can be enabled after :ref:`building and running the sample <matter_lock_sample_remote_control>`.
+
+.. _matter_weather_station_network_mode:
+
+Remote testing in a network
+===========================
+
+.. include:: ../lock/README.rst
+    :start-after: matter_door_lock_sample_remote_testing_start
+    :end-before: matter_door_lock_sample_remote_testing_end
+
+.. _matter_weather_station_test_mode:
+
+Remote testing using test mode
+==============================
+
+.. include:: ../lock/README.rst
+    :start-after: matter_door_lock_sample_test_mode_start
+    :end-before: matter_door_lock_sample_test_mode_end
+
+Configuration
+*************
+
+|config|
+
+Device Firmware Upgrade support
+===============================
+
+.. include:: ../lock/README.rst
+    :start-after: matter_door_lock_sample_build_with_dfu_start
+    :end-before: matter_door_lock_sample_build_with_dfu_end
+
+User interface
+**************
+
+.. include:: ../lock/README.rst
+    :start-after: matter_door_lock_sample_button1_start
+    :end-before: matter_door_lock_sample_button1_end
+
+Serial Wire Debug port:
+    Used for getting logs from the device or communicating with it through the command-line interface.
+
+NFC port with antenna attached:
+    Optionally used for obtaining the commissioning information from the Matter accessory device to start the :ref:`commissioning procedure <matter_weather_station_sample_remote_control_commissioning>`.
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/matter/weather_station`
+
+.. include:: /includes/build_and_run.txt
+
+See `Configuration`_ for information about building the sample with the DFU support.
+
+Testing
+=======
+
+You can either test the sample's :ref:`basic features <matter_light_bulb_sample_basic_features_tests>` or use the light switch sample to test the light bulb's :ref:`communication with another device <matter_light_bulb_sample_light_switch_tests>`.
```

Comment:
As discussed F2F for now I would mention that currently sample is still in development and cannot be tested, because we are lacking of proper mobile app with data visualization.